dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/importers/ui/ImportersForView.java

```diff
@@ -0,0 +1,82 @@
+package edu.harvard.iq.dataverse.importers.ui;
+
+import edu.harvard.iq.dataverse.importer.metadata.ImporterConstants;
+import edu.harvard.iq.dataverse.importer.metadata.MetadataImporter;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.MetadataBlock;
+import edu.harvard.iq.dataverse.persistence.dataverse.Dataverse;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.MissingResourceException;
+import java.util.ResourceBundle;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ImportersForView implements MapForView<MetadataImporter, ImportersForView.ImporterItem> {
+    private Map<MetadataImporter, ImporterItem> items = new HashMap<>();
+
+    // -------------------- GETTERS --------------------
+
+    public List<MetadataImporter> getImportersView() {
+        return new ArrayList<>(items.keySet()); // TODO: Implement some kind of sorting later
+    }
+
+    @Override
+    public Map<MetadataImporter, ImporterItem> getUnderlyingMap() {
+        return items;
+    }
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    public ImportersForView(Dataset dataset, Map<String, MetadataImporter> importers, Locale locale) {
+
+        if (dataset == null || locale == null) {
+            return;
```

Comment:
Do we really want to silently fail when clearly someone is using class api in not intended way?

Reply:
I meant the second.
Correct if i''m wrong but it doesn't seem to have anything to do about configuration issue. Rather invalid use of this class?

Reply:
Yes, you're right. I'll change it.